io/io/inc/TFileMergeClient.h

```diff
@@ -0,0 +1,37 @@
+#ifndef ROOT_TFileMergeClient
+#define ROOT_TFileMergeClient
+
+#include "TMemFile.h"
+
+class TArrayC;
+class TBufferFile;
+
+namespace ROOT {
+namespace Experimental {
+
+class TFileMergeServer;
+
+class TFileMergeClient : public TMemFile {
+private:
+   TFileMergeServer& fServer;
+   TArrayC *fClassSent;
+
+   TFileMergeClient(TFileMergeServer& s);
+
+   TFileMergeClient() = delete;
+   TFileMergeClient(const TFileMergeClient&) = delete;
+
+   friend class TFileMergeServer;
+
+public:
+   ~TFileMergeClient();
+
+   virtual Int_t Write(const char *name = nullptr, Int_t opt = 0, Int_t bufsize = 0);
+   virtual Int_t Write(const char *name = nullptr, Int_t opt = 0, Int_t bufsize = 0) const;
```

Comment:
I tried before, but got several warnings. I will do it if I can manage to find a way to do it without the warnings.